vars/daosLatestVersion.groovy

```diff
@@ -31,9 +31,14 @@ String distro2repo(String distro) {
 String getLatestVersion(String distro, BigDecimal next_version, String type='stable') {
     String v = null
     String repo = 'daos-stack-daos-' + distro2repo(distro) + '-x86_64-' + type + '-local/'
+    String artifactory_url = env.ARTIFACTORY_URL
+    /* For backwards support, add the 'artifactory' path if it is missing from the env */
+    if (!artifactory_url.endsWith('/artifactory')) {
+        artifactory_url = "${artifactory_url}/artifactory"
+    }
     try {
         v = sh(label: 'Get RPM packages version for: ' + repo + ' with version < ' + next_version.toString(),
-               script: '$(command -v dnf) --refresh repoquery --repofrompath=daos,' + env.ARTIFACTORY_URL +
+               script: '$(command -v dnf) --refresh repoquery --repofrompath=daos,' + artifactory_url + '/' +
                        repo + ''' --repoid daos --qf %{version}-%{release} --whatprovides 'daos < ''' +
```

Comment:
If we don't want/need to support an `env.ARTIFACTORY_URL` that does not end in `artifactory`, we could simplify this to:
```suggestion
    String repo_url = "${env.ARTIFACTORY_URL}/${repo}"
    try {
        v = sh(label: 'Get RPM packages version for: ' + repo + ' with version < ' + next_version.toString(),
               script: '$(command -v dnf) --refresh repoquery --repofrompath=daos,' + repo_url +
                       ''' --repoid daos --qf %{version}-%{release} --whatprovides 'daos < ''' +
```